BridgeAppSDK/SBAGenericStepDataSource.swift

```diff
@@ -0,0 +1,614 @@
+        if let delegate = delegate {
+            delegate.answersDidChange()
+        }
+    }
+    
+    /**
+     Retrieve the current ORKStepResult.
+     @return    An ORKStepResult object with the current answers for all of its ORKFormItems
+     */
+    open func results(parentResult: ORKStepResult) -> ORKStepResult {
+        
+        if let formItems = formItemsWithAnswerFormat() {
+            
+            // "Now" is the end time of the result, which is either actually now,
+            // or the last time we were in the responder chain.
+            
+            let now = parentResult.endDate
+            var qResults: Array<ORKResult> = Array()
+            
+            for formItem: ORKFormItem in formItems {
+                
+                
+                // Skipped forms report a "null" value for every item -- by skipping, the user has explicitly said they don't want
+                // to report any values from this form.
+                
+                var answer = ORKNullAnswerValue()
+                var answerDate = now
+                var systemCalendar = Calendar.current
+                var systemTimeZone = NSTimeZone.system
+                
+                // TODO: Josh Bruhin, 6/12/17 - implement skipped?
+                let skipped = false
+                if !skipped {
+                    
+                    if let itemGroup = itemGroup(with: formItem.identifier) {
+                        answer = itemGroup.answer
+                        answerDate = itemGroup.answerDate ?? now
+                        systemCalendar = itemGroup.calendar
+                        systemTimeZone = itemGroup.timezone
+                    }
+                }
+                
+                let result = formItem.answerFormat?.result(withIdentifier: formItem.identifier, answer: answer)
+                let impliedAnswerFormat = formItem.answerFormat?.implied()
+                
+                if let dateAnswerFormat = impliedAnswerFormat as? ORKDateAnswerFormat, let dateQuestionResult = result as? ORKDateQuestionResult {
+                    if let _ = dateQuestionResult.dateAnswer {
+                        let usedCalendar = dateAnswerFormat.calendar ?? systemCalendar
+                        dateQuestionResult.calendar = usedCalendar
+                        dateQuestionResult.timeZone = systemTimeZone
+                    }
+                } else if let numericAnswerFormat = impliedAnswerFormat as? ORKNumericAnswerFormat {
+                    if let numericQuestionFormat = result as? ORKNumericQuestionResult {
+                        numericQuestionFormat.unit = numericAnswerFormat.unit
+                    }
+                }
+                
+                result?.startDate = answerDate
+                result?.endDate = answerDate
+                
+                qResults.append(result!)
+            }
+            
+            parentResult.results = parentResult.results! + qResults
```

Comment:
Makes sense. There are a few sections in this class where I copied from RK, I'll have to make them swifty.